roles/splunk/tasks/configure_license.yml

```diff
@@ -1,4 +1,33 @@
 ---
+- block:
+  - name: Copy Licenses to Master
+    copy:
+      src: "{{ item }}"
+      dest: "/tmp/{{ item }}"
+    loop: "{{ splunk_licenses }}"
+    
+  - name: Install license on licensemaster
+    command: "{{ splunk_home }}/bin/splunk add licenses /tmp/{{ item }} -auth {{ splunk_auth }}"
+    become: true
+    become_user: "{{ splunk_nix_user }}"
+    loop: "{{ splunk_licenses }}"
+    ignore_errors: yes
+    register: license_result
+    changed_when: '"The licenses object has been added" in license_result.stdout'
+    notify: restart splunk
+
+  - name: Clean Up licenses
+    file:
+      path: "/tmp/{{ item }}"
+      state: absent
+    become: true
+    loop: "{{ splunk_licenses }}"
+
+  when:
+    - "'licensemaster' in group_names"
+    - splunk_licenses is defined
```

Comment:
This line isn't needed because you are already defining `splunk_licenses` in defaults/main.yml